simapp/simd/cmd/genaccounts.go

```diff
@@ -72,119 +68,18 @@ contain valid denominations. Accounts may optionally be supplied with vesting pa
 				}
 			}
 
-			coins, err := sdk.ParseCoinsNormalized(args[1])
-			if err != nil {
-				return fmt.Errorf("failed to parse coins: %w", err)
-			}
-
+			appendflag, _ := cmd.Flags().GetBool(flagAppendMode)
 			vestingStart, _ := cmd.Flags().GetInt64(flagVestingStart)
 			vestingEnd, _ := cmd.Flags().GetInt64(flagVestingEnd)
 			vestingAmtStr, _ := cmd.Flags().GetString(flagVestingAmt)
 
-			vestingAmt, err := sdk.ParseCoinsNormalized(vestingAmtStr)
-			if err != nil {
-				return fmt.Errorf("failed to parse vesting amount: %w", err)
-			}
-
-			// create concrete account type based on input parameters
-			var genAccount authtypes.GenesisAccount
-
-			balances := banktypes.Balance{Address: addr.String(), Coins: coins.Sort()}
-			baseAccount := authtypes.NewBaseAccount(addr, nil, 0, 0)
-
-			if !vestingAmt.IsZero() {
-				baseVestingAccount := authvesting.NewBaseVestingAccount(baseAccount, vestingAmt.Sort(), vestingEnd)
-
-				if (balances.Coins.IsZero() && !baseVestingAccount.OriginalVesting.IsZero()) ||
-					baseVestingAccount.OriginalVesting.IsAnyGT(balances.Coins) {
-					return errors.New("vesting amount cannot be greater than total amount")
-				}
-
-				switch {
-				case vestingStart != 0 && vestingEnd != 0:
-					genAccount = authvesting.NewContinuousVestingAccountRaw(baseVestingAccount, vestingStart)
-
-				case vestingEnd != 0:
-					genAccount = authvesting.NewDelayedVestingAccountRaw(baseVestingAccount)
-
-				default:
-					return errors.New("invalid vesting parameters; must supply start and end time or end time")
-				}
-			} else {
-				genAccount = baseAccount
-			}
-
-			if err := genAccount.Validate(); err != nil {
-				return fmt.Errorf("failed to validate new genesis account: %w", err)
-			}
-
-			genFile := config.GenesisFile()
-			appState, genDoc, err := genutiltypes.GenesisStateFromGenFile(genFile)
-			if err != nil {
-				return fmt.Errorf("failed to unmarshal genesis state: %w", err)
-			}
-
-			authGenState := authtypes.GetGenesisStateFromAppState(clientCtx.Codec, appState)
-
-			accs, err := authtypes.UnpackAccounts(authGenState.Accounts)
-			if err != nil {
-				return fmt.Errorf("failed to get accounts from any: %w", err)
-			}
-
-			bankGenState := banktypes.GetGenesisStateFromAppState(clientCtx.Codec, appState)
-			if accs.Contains(addr) {
-				appendflag, _ := cmd.Flags().GetBool(flagAppendMode)
-				if !appendflag {
-					return fmt.Errorf("cannot add account at existing address %s", addr)
-				}
-
-				genesisB := banktypes.GetGenesisStateFromAppState(clientCtx.Codec, appState)
-				for idx, acc := range genesisB.Balances {
-					if acc.Address != addr.String() {
-						continue
-					}
-
-					updatedCoins := acc.Coins.Add(coins...)
-					bankGenState.Balances[idx] = banktypes.Balance{Address: addr.String(), Coins: updatedCoins.Sort()}
-					break
-				}
+			if filepath.IsAbs(config.Genesis) {
```

Comment:
Why this special change for handling genesis path?

Reply:
because `config.Genesis` returns relative path, this was just to make sure we have full absolute path was given to `AddGenesisAccount` func.

Reply:
thanks for pointing this @alexanderbez , I could indeed use `config.GenesisFile()` instead of all these lines, my bad :(